source/extensions/clusters/logical_dns/logical_dns_cluster.h

```diff
@@ -85,15 +93,22 @@ class LogicalDnsCluster : public ClusterImplBase {
   const envoy::config::endpoint::v3::ClusterLoadAssignment load_assignment_;
 };
 
-class LogicalDnsClusterFactory : public ClusterFactoryImplBase {
+class LogicalDnsClusterFactory : public Upstream::ConfigurableClusterFactoryBase<
+                                     envoy::extensions::clusters::dns::v3::DnsCluster> {
```

Comment:
So confusingly, inheriting from `ConfigurableClusterFactoryBase` does not do any additional registration. This inheritance only adds some helper methods that parse out the `typed_config`. The inheritance hierarchy is `*DnsClusterFactory -> ConfigurableClusterFactoryBase -> ClusterFactoryImplBase -> ClusterFactory -> UntypedFactory`. The type registratrion is usually done with `TypedFactory` which inherits from `UntypedFactory`. That being said, we don't have to use `TypedFactory` for this. I just though it was enough of a self contained change that we could do it in another PR, but I can do it here too.

Reply:
Interesting (and confusing)!  I agree that it would be good to clean this up and make it work more like the other extension types, which I guess means using `TypedFactory`.

Reply:
This is what I want! But many factories don't have an extension type and so we can't have `ClusterFactory` inherit from `TypedFactory`, but this point is more of a implementation detail and I can work around it.